### PR TITLE
checking for policy only mode

### PR DESCRIPTION
**Pull request by heschlie**
## Description

Fixes projectcalico/calicoctl#1597
This PR will stop `startup.go` from writing IP and ASnum data to the Node when Calico is run in policy only mode.

## Todos

- [ ] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
calico/node will no longer write BGP related info to the datastore when run in policy only mode.  This allows calico/node to run with reduced permissions in policy only mode.
```

### CODE REVIEW

**Comment by heschlie** [comment]
Still sorting out ST failures...

File: calico_node/startup/startup.go [hunk 1]
```diff
@@ -86,18 +86,20 @@ func main() {
 	// updated IP data and use the full list of nodes for validation.
 	node := getNode(client, nodeName)
 
-	// Configure and verify the node IP addresses and subnets.
-	checkConflicts := configureIPsAndSubnets(node)
+	if os.Getenv("CALICO_NETWORKING_BACKEND") == "none" {
```
Comment (gunjan5) [inline calico_node/startup/startup.go]: shouldn't this be `!=`? right now it's configuring IP and checking for conflict when it's in policy only mode. Don't we want the opposite?

Comment (gunjan5) [inline calico_node/startup/startup.go]: Also, can you add a comment before this line so we know why we're skipping setting IP and checking conflict?

Comment (heschlie) [inline calico_node/startup/startup.go]: doh, this is wrong, fixed locally and re-running STs

**Comment by caseydavenport** [comment]
This LGTM but I'd like @robbrockbank to do a quick sanity check.

**Comment by heschlie** [comment]
ping @gunjan5 @robbrockbank

**Comment by gunjan5** [comment]
@heschlie what's with the ST failures?

**Comment by heschlie** [comment]
Not sure, I'm not seeing it fail locally, I kicked off another run.

**Comment by robbrockbank** [comment]
Don't forget to squash the commits.

File: calico_node/startup/startup.go [hunk 2]
```diff
@@ -118,7 +122,7 @@ func main() {
 
 	// Write the startup.env file now that we are ready to start other
 	// components.
-	writeStartupEnv(nodeName, node.Spec.BGP.IPv4Address, node.Spec.BGP.IPv6Address)
```
Comment (gunjan5) [inline calico_node/startup/startup.go]: 👍

**Pull request merged**